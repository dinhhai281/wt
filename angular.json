{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "projects",
  "projects": {
    "angular-unit-test": {
      "projectType": "application",
      "schematics": {
        "@schematics/angular:component": {
          "style": "scss",
          "changeDetection": "OnPush",
          "viewEncapsulation": "None"
        }
      },
      "root": "",
      "sourceRoot": "src",
      "prefix": "app",
      "architect": {
        "build": {
          "builder": "@angular-builders/custom-webpack:browser",
          "options": {
            "outputPath": "dist/angular-unit-test",
            "index": "src/index.html",
            "main": "src/main.ts",
            "polyfills": "src/polyfills.ts",
            "tsConfig": "tsconfig.app.json",
            "aot": true,
            "assets": [
              "src/favicon.ico",
              "src/assets"
            ],
            "styles": [
              "src/styles.scss"
            ],
            "scripts": [],
            "customWebpackConfig": {
              "path": "webpack.config.js"
            }
          },
          "configurations": {
            "production": {
              "fileReplacements": [
                {
                  "replace": "src/environments/environment.ts",
                  "with": "src/environments/environment.prod.ts"
                }
              ],
              "optimization": true,
              "outputHashing": "all",
              "sourceMap": false,
              "extractCss": true,
              "namedChunks": false,
              "extractLicenses": true,
              "vendorChunk": false,
              "buildOptimizer": true,
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "2mb",
                  "maximumError": "5mb"
                },
                {
                  "type": "anyComponentStyle",
                  "maximumWarning": "6kb",
                  "maximumError": "10kb"
                }
              ]
            }
          }
        },
        "serve": {
          "builder": "@angular-builders/custom-webpack:dev-server",
          "options": {
            "browserTarget": "angular-unit-test:build",
            "customWebpackConfig": {
              "path": "webpack.config.js"
            }
          },
          "configurations": {
            "production": {
              "browserTarget": "angular-unit-test:build:production"
            }
          }
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "browserTarget": "angular-unit-test:build"
          }
        },
        "lint": {
          "builder": "@angular-eslint/builder:lint",
          "options": {
            "lintFilePatterns": [
              "src/**/*.ts",
              "src/**/*.component.html"
            ],
            "exclude": [
              "**/node_modules/**"
            ]
          }
        }
      }
    }
  },
  "defaultProject": "angular-unit-test"
}
