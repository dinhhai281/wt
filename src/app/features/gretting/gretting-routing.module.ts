import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';

import { GrettingComponent } from './gretting.component';

const routes: Routes = [{ path: '', component: GrettingComponent }];

@NgModule({
  imports: [RouterModule.forChild(routes)],
  exports: [RouterModule],
})
export class GrettingRoutingModule {
  static components = [GrettingComponent];
}
