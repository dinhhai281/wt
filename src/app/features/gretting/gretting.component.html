<p class="bg-red-500">gretting works!</p>
