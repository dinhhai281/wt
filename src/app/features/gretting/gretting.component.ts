import { Component, OnInit, ViewEncapsulation, ChangeDetectionStrategy } from '@angular/core';

@Component({
  selector: 'app-gretting',
  templateUrl: './gretting.component.html',
  styleUrls: ['./gretting.component.scss'],
  encapsulation: ViewEncapsulation.None,
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class GrettingComponent implements OnInit {
  constructor() {}

  ngOnInit(): void {}
}
