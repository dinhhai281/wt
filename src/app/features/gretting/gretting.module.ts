import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { GrettingRoutingModule } from './gretting-routing.module';

import { GrettingComponent } from './gretting.component';

@NgModule({
  declarations: [...GrettingRoutingModule.components],
  imports: [CommonModule, GrettingRoutingModule],
})
export class GrettingModule {}
