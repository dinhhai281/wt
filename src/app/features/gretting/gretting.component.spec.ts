import { render, RenderResult } from '@testing-library/angular';

import { GrettingComponent } from './gretting.component';

describe('GrettingComponent', () => {
  let component: RenderResult<GrettingComponent>;

  beforeEach(async () => {
    component = await render(GrettingComponent);
  });

  it('should match snapshot', () => {
    expect(component.container).toMatchSnapshot();
  });
});
