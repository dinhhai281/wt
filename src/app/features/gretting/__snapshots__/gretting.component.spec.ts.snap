// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GrettingComponent should match snapshot 1`] = `
<div
  id="root0"
>
  <p
    class="bg-red-500"
  >
    gretting works!
  </p>
</div>
`;
